Assignment/Assignment.Tests/PingProcessTests.cs

```diff
@@ -59,42 +60,59 @@ public void RunTaskAsync_Success()
     {
         // Do NOT use async/await in this test.
         // Test Sut.RunTaskAsync("localhost");
+        Task<PingResult> result = Sut.RunTaskAsync("localhost");
+        AssertValidPingOutput(result.Result);
     }
 
     [TestMethod]
     public void RunAsync_UsingTaskReturn_Success()
     {
         // Do NOT use async/await in this test.
-        PingResult result = default;
         // Test Sut.RunAsync("localhost");
-        AssertValidPingOutput(result);
+        Task<PingResult> result = Sut.RunAsync("localhost");
+        AssertValidPingOutput(result.Result);
     }
 
     [TestMethod]
-#pragma warning disable CS1998 // Remove this
     async public Task RunAsync_UsingTpl_Success()
     {
-        // DO use async/await in this test.
-        PingResult result = default;
+        
+        PingResult result = await Sut.RunAsync("localhost");
 
-        // Test Sut.RunAsync("localhost");
+       
         AssertValidPingOutput(result);
     }
-#pragma warning restore CS1998 // Remove this
 
 
     [TestMethod]
     [ExpectedException(typeof(AggregateException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
     {
-        
+        CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
+        cancellationTokenSource.Cancel(); 
+        Task<PingResult> result = Sut.RunAsync("localhost",  cancellationToken);
+        AssertValidPingOutput(result.Result);
     }
 
     [TestMethod]
     [ExpectedException(typeof(TaskCanceledException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrappingTaskCanceledException()
     {
-        // Use exception.Flatten()
+        
+        CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
+        CancellationToken cancellationToken = cancellationTokenSource.Token;
+        cancellationTokenSource.Cancel();
+        Task<PingResult> result = Sut.RunAsync("localhost", cancellationToken);
+        try
+        {
+            AssertValidPingOutput(result.Result);
+        }
+        catch(AggregateException ex)
+        {
+            Exception taskCanceledException = ex.Flatten();
+            throw taskCanceledException.InnerException!;
+        }

```

Comment:
Again here you want to be calling Cancel after the call Sut.RunAsync but if you do that your test won't pass so you'll need to fix some other things. 

```suggestion
        CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
        CancellationToken cancellationToken = cancellationTokenSource.Token;
        Task<PingResult> result = Task.Run(() => Sut.RunAsync("localhost", cancellationToken));
        cancellationTokenSource.Cancel();
        try
        {
            _ = result.Result;
        }
        catch(AggregateException ex)
        {
            Exception taskCanceledException = ex.Flatten();
            throw taskCanceledException.InnerException!;
        }
```